### ISSUE TITLE
Configure Puppeteer's page-level interaction timeout

### ISSUE THREAD

**Issue by jonscottclark** [garris/BackstopJS#1048]
Hi @garris, I'm wondering if we could introduce an option to configure the default timeout for page interactions, that would normally be set directly in Puppeteer `page` methods as a timeout option.

Since we are allowing `waitTimeout` to be passed in the `engineOptions` property (despite being a non-valid property that `puppeteer` would accept directly) and having it used here: https://github.com/garris/BackstopJS/blob/b45fab38661056ad94abff26b2ee87ff0157a8d0/core/util/runPuppet.js#L67

Can we add an option and potentially pass that value to  `page.setDefaultTimeout()`, and place it after the line pasted above? Then we can set a timeout higher than 30 seconds for any page-level interactions like waiting for a selector?

Thank you for your work on this, and I look forward to hearing your feedback!

**Comment by garris** [comment]
Hi, Thanks for posting!   So, just want to make sure I understand -- I think you just want to add another line after line 67 above 👆 ... like this? ```
page.setDefaultTimeout(probably_use_the_same_waitTimeout_value_here)
```
Like that?

**Comment by jonscottclark** [comment]
Yes, that's the idea! Sometimes when running against a very slow server, even though we navigate to the page in time, waiting for the `readySelector` to display takes longer than 30 seconds, and so it would be great to be able to configure that.

Ideally it would be a separate configuration option - though I'm not sure it makes sense to have it in `engineOptions` because it's not an official property that would be passed to `puppeteer.launch()`, this new one and `waitTimeout` are more like options that Puppeteer would use further down in its processing. Since we probably don't want to move `waitTimeout` (that would break people's configurations), it makes sense to use `waitTimeout` again for `page.setDefaultTimeout()` as you suggest, or create a brand new property like `pageWaitTimeout`.

**Comment by garris** [comment]
Yes, I think that one was a legacy prop from when we were using casper/phantom.

It would be way better to have `waitTimeout` be declared at the scenario (and/or root) level as a first-class property.   (while continuing to check if its used in the existing place for compatibility)

Would be great if you could put in a PR for this!

**Comment by jonscottclark** [comment]
@garris Your suggestion to move it to a scenario level (while still letting it be globally set) is good. Would you use that timeout value for both 1) initial navigation and 2) ready event, or should there be a new property added for the ready event (since `waitTimeout` is currently only concerned with the navigation timeout).

**Comment by garris** [comment]
@jonscottclark -- I am leaning towards keeping the one property for both.   I will probably regret any decision I make at this point so, I would like to apologize to you and my future-self now for any short-sighted thinking on this.  😉
